systems/data-plan/api-gateway/pkg/rest/router.go

```diff
@@ -124,14 +125,15 @@ func (p *Router) getPackagesHandler(c *gin.Context) (*pb.GetByOrgPackageResponse
 	return resp, nil
 }
 func (p *Router) getBaseRateHandler(c *gin.Context) (*pbBaseRate.GetBaseRateResponse, error) {
-	rateId, error := strconv.ParseUint(c.Param("baseRate"), 10, 64)
-	if error != nil {
-		logrus.Error(error)
+	rateID, err := uuid.FromString(c.Param("baseRate"))
+
+	if err != nil {
+		logrus.Error(err)
 		return nil, &rest.HttpError{HttpCode: http.StatusBadRequest,
 			Message: "baseRate is not valid!"}
 	}
 	resp, err := p.clients.d.GetBaseRate(&pbBaseRate.GetBaseRateRequest{
-		RateId: rateId,
+		RateID: rateID.String(),
```

Comment:
should this be UUID ?

Reply:
It is a UUID, but since PB does not recognize UUID, it is treated as a string. However, the validation process will ensure that it conforms to the UUID format.